docs/proposal-new-output-syntax.md

```diff
@@ -0,0 +1,769 @@
+loop (use execution timeout) or if it never started running in the first place.
+
+The new optional output syntax cannot be used with family collective psuedo
+output triggers because it is difficult to interpret that in terms of the
+success status of each individual member, which is what we need.
+Instead we have different optional/required defaults for each collective
+pseudo-output. See [Family Triggers](#Family-Triggers) below. 
+
+## Path Branching
+
+### Concurrent Paths 
+
+Concurrent paths can trigger off of required outputs:
+```
+a:x => b1
+a:y => b2
+b1 & b2 => c  # join
+```
+
+### Alternate Paths
+
+Alternate paths must branch off of optional outputs, and must be joined with
+OR triggers because only one side or the other will run:
+
+```
+a:x? => b1
+a:y? => b2
+b1 | b2 => c  # join
+```
+Note joining can trigger off of required outputs, because the output is
+only "required" if the task actually runs.
+
+**Failure recovery** is a common example of this:
+```
+a? => b1  # success path
+a:fail? => b2  # failure path
+b1 | b2 => c  # join
+```
+
+Note that an artificial dependency can be used to *ensure that at least one
+of the alternate paths gets taken*:
+```
+a:x? => b1
+a:y? => b2
+b1 | b2 => c  # join
+a => c # ARTIFICIAL DEPENDENCY
+```
+This forces `c` to be spawned as an unsatisfied prerequisite waiting on `b1` or
+`b2` even if neither branch runs. If the alternate paths don't actually need to
+join, `c` can be a dummy task.
+
+*See Appendix Output Completion Expressions for a possible follow-up that
+would allow us to flag `a` as incomplete here and avoid the artificial
+dependency.*
+
+## Failed Tasks
+
+No special handling of failed tasks is required.
+- if success is required, a failed task automatically gets retained in the task
+  pool as incomplete.
+- if success is optional (which would usually imply that optional failure is
+  explicitly handled), a failed task can be removed as complete
+
+No special syntax (e.g. `c:fail => NULL`) or suicide triggers (e.g. `c:fail =>
+!c`) are needed for leaf tasks either, if we don't care if they succeed or fail:
+```
+a => b => c?  # success of c is optional
+```
+
+## Family Triggers
+
+The new optional output syntax cannot be used with family collective psuedo
+output triggers because it is difficult to interpret that in terms of the
+success status of each individual member, which is what we need.
+
+Instead we will set different optional/required defaults for each collective
+pseudo output:
+
+```
+A:succeed-all => b  # member success required
+A:succeed-any => b  # member success/failure optional
+
+A:fail-all => b  # member failure required 
+A:fail-any => b  # member success/failure optional
+
+A:finish-all => b  # member success optional
+A:finish-any => b  # member success optional
+
+A:start-all => b  # N/A
+A:start-any => b  # N/A
+
+A:submit-all => b  # member submit success required
+A:submit-any => b  # member success optional
```

Comment:
"success" should be "submit/submit-fail"

Reply:
:+1:  Oops, cut-n-paste fail.